### DISCUSSION TITLE
有关代码的相关问题，对前端代码修改的讨论

### DISCUSSION THREAD

**Discussion by PairZhu** [jeessy2/ddns-go#985] [General]
我正在准备提交一个新的PR，大概目标如下：
1. 优化前端代码结构
2. 优化前端的国际化
3. 增加正则和命令测试
4. 小幅修改UI(可能)
但是阅读新增的代码时有一些疑问，想请教一下大佬们，此外还有一些修改想听听大佬们的意见，如果方便的话可以是否可以有更方便的和大佬们交流的联系方式，如果担心隐私问题不愿意留联系方式可以查看我的profile联系我，又或者实在觉得不妥就在这个discussion中讨论也是可以的。
我这边遇到的一个疑惑点在于，为什么这两个配置项需要和语言有关：
https://github.com/jeessy2/ddns-go/blob/fe9f437d9b1c600bdafd5af9c15f2ed9ce193538/web/writing.html#L792-L794
https://github.com/jeessy2/ddns-go/blob/fe9f437d9b1c600bdafd5af9c15f2ed9ce193538/web/writing.html#L799-L801

**Comment by jeessy2** [comment]
欢迎PR
英文地区优先使用国外的接口获取IP是比较合适的

**Comment by PairZhu** [comment]
这里对于中文版本增加一层p标签的作用是什么？原来的版本好像是没有的
https://github.com/jeessy2/ddns-go/blob/fe9f437d9b1c600bdafd5af9c15f2ed9ce193538/web/writing.html#L1202-L1207

**Reply by jeessy2** [reply]
可以不要

**Comment by PairZhu** [comment]
**请求添加JQuery库**
JQuery库的大小只有30kB不到，远小于已有的bootstrap库，我原计划手动实现jQuery的所需要的功能，但是发现不是很划算，可能带来各种bug和隐患。加入jQuery之后可以很明显的缩小业务代码的体积代码逻辑更加简洁。  
并且jQuery等价于原生dom操作，并没有引入新的模型，原代码结构变化不会很剧烈，更不会带来性能问题

**Reply by jeessy2** [reply]
jQuery与Bootstrap是两个不同功能的库，以前使用了jQuery，后面去掉了。原生的js也是可以的，使用jQuery也减少不了多少代码

**Reply by PairZhu** [reply]
我的意思是你都不在意Bootstrap的体积了，那么更没必要在意jQuery的体积了，而且我这边感觉确实用了jQuery之后代码清爽了很多

**Reply by PairZhu** [reply]
少了很多繁杂的dom操作，我到时候提交pr之后你看看效果吧

**Reply by jeessy2** [reply]
以前jQuery改成原生js，也感觉清爽了一些。。。现在用不到jQuery啥特性吧？

**Reply by PairZhu** [reply]
jQuery对css和className的操作，onload，click时间，选择器这几个在目前的代码里频繁出现，而且经常导致代码宽度过大而需要折行

**Reply by jeessy2** [reply]
你可以看下之前的提交#475。改成jQuery也好不到哪里去，可能还会引起其它问题

**Reply by PairZhu** [reply]
我没看出来这个会导致其它什么问题，只是提到了减少静态资源大小呀？但是这点大小其实不值一提

**Reply by jeessy2** [reply]
少的改动引起的bug少，大的会引出更多的bug。文件还是减小了80KB

**Reply by PairZhu** [reply]
使用的版本不对吧，我这边的jquery只有27kB

**Reply by jeessy2** [reply]
之前3版本，80KB。长点就长点吧，运行正常就行，又不是看不懂，会原生js也是一种能力

**Reply by PairZhu** [reply]
此外，使用jquery代替原生js并不容易带来bug，因为jquery本质是只是个语法糖

**Reply by jeessy2** [reply]
我指的人为问题，不是框架。

**Reply by jeessy2** [reply]
你觉得不好看，根据不同功能单独创建不同的js文件

**Comment by PairZhu** [comment]
是否可以把网卡信息获取api删除，改成和其它数据一样在请求网页时插值进html内

**Reply by jeessy2** [reply]
如果没有什么明显的优势的话，尽量保持不变

**Comment by PairZhu** [comment]
如果增加正则测试，是直接用js代码进行测试减少通讯频率比较好，还是交给后端保证结果的一致性比较好

**Reply by jeessy2** [reply]
保证一致性比较好
